### PR TITLE
Fixes to the release script

### PR DESCRIPTION
**Pull request by babolivier**
* rename major/minor into the right semver terminology minor/patch (since this was something that got me very confused the first couple of times I've used the script)
* name the release branch based on the new version, not the previous one

### CODE REVIEW

File: scripts-dev/release.py [hunk 1]
```diff
@@ -97,49 +91,57 @@ def run():
...
         else:
-            new_version = base_version
+            new_version = "{}.{}.{}".format(
+                current_version.major,
+                current_version.minor,
+                current_version.micro,
+            )
     else:
         # If this is a new release cycle then we need to know if its a major
         # version bump or a hotfix.
```
Comment (anoadragon453) [inline scripts-dev/release.py:101]:
```suggestion
        # If this is a new release cycle then we need to know if it's a minor
        # or a patch version bump.
```

File: changelog.d/10239.misc [hunk 2]
```diff
@@ -0,0 +1 @@
+Fixes to the release script.
```
Comment (anoadragon453) [inline changelog.d/10239.misc:1]: This lacks some description of what the changes actually were..

**Review by anoadragon453 — APPROVED** [review]
Thanks for doing this!

File: scripts-dev/release.py (hunk 1 above)
Comment (babolivier) [reply scripts-dev/release.py:101]: Oh, good catch!

**Review by anoadragon453 — APPROVED** [review]
lgtm!

**Pull request merged**